File: go-client-sdk/pkg/models/shared/codatpublicapimodelsdatadataset.go
package shared

import (
	"time"
)

type CodatPublicAPIModelsDataDataSet struct {
	CompanyID                string                                    `json:"companyId"`
	Completed                *time.Time                                `json:"completed,omitempty"`
	ConnectionID             string                                    `json:"connectionId"`
	DataType                 *string                                   `json:"dataType,omitempty"`
	ErrorMessage             *string                                   `json:"errorMessage,omitempty"`
	ID                       string                                    `json:"id"`
	IsCompleted              bool                                      `json:"isCompleted"`
	IsErrored                bool                                      `json:"isErrored"`
	Progress                 int32                                     `json:"progress"`
	Requested                time.Time                                 `json:"requested"`
	Status                   CodatPublicAPIModelsDataDatasetStatusEnum `json:"status"`
	ValidationinformationURL *string                                   `json:"validationinformationUrl,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsdatadataset.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatPublicApiModelsDataDatasetStatusEnum } from "./codatpublicapimodelsdatadatasetstatusenum";
import { Expose, Transform } from "class-transformer";


export class CodatPublicApiModelsDataDataSet extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "companyId" })
  companyId: string;

  @SpeakeasyMetadata()
  @Expose({ name: "completed" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  completed?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "connectionId" })
  connectionId: string;

  @SpeakeasyMetadata()
  @Expose({ name: "dataType" })
  dataType?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "datasetLogsUrl" })
  datasetLogsUrl?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "errorMessage" })
  errorMessage?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "id" })
  id: string;

  @SpeakeasyMetadata()
  @Expose({ name: "isCompleted" })
  isCompleted: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "isErrored" })
  isErrored: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "progress" })
  progress: number;

  @SpeakeasyMetadata()
  @Expose({ name: "requested" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  requested: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status: CodatPublicApiModelsDataDatasetStatusEnum;

  @SpeakeasyMetadata()
  @Expose({ name: "validationinformationUrl" })
  validationinformationUrl?: string;
}